src/coreclr/jit/importer.cpp

```diff
@@ -4235,6 +4235,24 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
             break;
         }
 
+        case CORINFO_FIELD_STATIC_DATASEGMENT:
+        {
+#ifdef FEATURE_READYTORUN
+            assert(opts.IsReadyToRun());
+            assert((pFieldInfo->fieldFlags & CORINFO_FLG_FIELD_INITCLASS) == 0);
```

Comment:
Why assert `CORINFO_FLG_FIELD_INITCLASS == 0`? I would expect the Jit to insert the necessary helper for it for these statics as well.